route-builder.js

```diff
@@ -0,0 +1,75 @@
+"use strict";
+var fs = require("fs");
+var cachePath = __dirname + '/route-cache.json';
+
+/*
+(if the route cache.json does not exist)
+    create the route-cache.json file
+    create a routes object with get, post etc objects in it
+    build our new route object
+    search through the routes object for the appropriate method,
+    then attach the route object
+    write the routes object back to the json object
+
+    return routes object
+
+if(check to see if route cache.json exists, if it does)
+    read from the file and attach the json object from the file to a variable in the builder called routes
+    build our new route object
+    search through the routes object for the appropriate method, then attach the route object
+    write the routes object back to route-cache.json
+
+    return routes object
+
+
+
+*/
+
+var routeBuilder = module.exports = exports = function(){
+  this.routes = {
+    'GET': {},
+    'POST':{},
+    'PUT':{},
+    'PATCH':{},
+    'DELETE':{}
+  };
+};
+
+routeBuilder.prototype.set = function(route, method, cb) {
+  if (routes["method"] == undefined){
+    console.log("The method does not exist. Please choose another.");
+
+  }else{
+    fs.stat(cachePath, function(error, stats){
+      if (error != null){
+        console.log("route-cache does not exist");
+        fs.writeFile(indexPath, function(err){
+          if (err) throw err;
+          console.log("route-cache.js was created");
+        });
+
+       this.routes['GET'][route] = cb;
+       var writeStream = fs.createWriteStream(cachePath);
```

Comment:
line 51 sets the route property to be the callback, so what is the writeStream's purpose?  I see you're writing it to routes, but what is it being returned to?  If a function wants to make use of this in the future, it could just reference the routes[method][route] property.